### PR TITLE
[SPARK-22423][SQL] Scala test source files like TestHiveSingleton.scala should be in scala source root

### PR DESCRIPTION
**Pull request by xubo245**
## What changes were proposed in this pull request?

  Scala test source files like TestHiveSingleton.scala should be in scala source root

## How was this patch tested?

Just move scala file from java directory to scala directory
No new test case in this PR.

```
	renamed:    mllib/src/test/java/org/apache/spark/ml/util/IdentifiableSuite.scala -> mllib/src/test/scala/org/apache/spark/ml/util/IdentifiableSuite.scala
	renamed:    streaming/src/test/java/org/apache/spark/streaming/JavaTestUtils.scala -> streaming/src/test/scala/org/apache/spark/streaming/JavaTestUtils.scala
	renamed:    streaming/src/test/java/org/apache/spark/streaming/api/java/JavaStreamingListenerWrapperSuite.scala -> streaming/src/test/scala/org/apache/spark/streaming/api/java/JavaStreamingListenerWrapperSuite.scala
	renamed:   sql/hive/src/test/java/org/apache/spark/sql/hive/test/TestHiveSingleton.scala  sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHiveSingleton.scala
```

### CODE REVIEW

**Comment by srowen** [comment]
See my comment on the JIRA. Let's fix all four instances at once. (And update the PR title)

**Comment by xubo245** [comment]
Ok, I will fix all and update the PR title Later.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHiveSingleton.scala [hunk 1]
```diff
@@ -24,7 +24,6 @@ import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.hive.HiveExternalCatalog
 import org.apache.spark.sql.hive.client.HiveClient
 
-
```
Comment (srowen) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHiveSingleton.scala]: Don't modify the file, just move it.

Comment (xubo245) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHiveSingleton.scala]: ok

**Comment by xubo245** [comment]
I have fixed all four instances and update the PR title.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**